.github/labeler.yml

```diff
@@ -18,109 +18,217 @@
 #
 
 INFRA:
-  - ".github/**/*"
-  - "appveyor.yml"
-  - "tools/**/*"
-  - "dev/create-release/**/*"
-  - ".asf.yaml"
-  - ".gitattributes"
-  - ".gitignore"
-  - "dev/merge_spark_pr.py"
-  - "dev/run-tests-jenkins*"
+  - changed-files:
+    - any-glob-to-any-file: [
+     '.github/**/*',
+     'appveyor.yml',
+     'tools/**/*',
+     'dev/create-release/**/*',
+     '.asf.yaml',
+     '.gitattributes',
+     '.gitignore',
+     'dev/merge_spark_pr.py',
+     'dev/run-tests-jenkins*'
+    ]
+
 BUILD:
- - any: ['dev/**/*', '!dev/merge_spark_pr.py', '!dev/run-tests-jenkins*']
- - "build/**/*"
- - "project/**/*"
- - "assembly/**/*"
- - "**/*pom.xml"
- - "bin/docker-image-tool.sh"
- - "bin/find-spark-home*"
- - "scalastyle-config.xml"
+  - changed-files:
+    - all-globs-to-any-file: [
+     'dev/**/*',
+     '!dev/merge_spark_pr.py',
+     '!dev/run-tests-jenkins*'
+    ]
+    - any-glob-to-any-file: [
+     'build/**/*',
+     'project/**/*',
+     'assembly/**/*',
+     '**/*pom.xml',
+     'bin/docker-image-tool.sh',
+     'bin/find-spark-home*',
+     'scalastyle-config.xml'
+    ]
+
 DOCS:
-  - "docs/**/*"
-  - "**/README.md"
-  - "**/CONTRIBUTING.md"
-  - "python/docs/**/*"
+  - changed-files:
+    - any-glob-to-any-file: [
+     'docs/**/*',
+     '**/README.md',
+     '**/CONTRIBUTING.md',
+     'python/docs/**/*'
+    ]
+
 EXAMPLES:
-  - "examples/**/*"
-  - "bin/run-example*"
+  - changed-files:
+    - any-glob-to-any-file: [
+     'examples/**/*',
+     'bin/run-example*'
+    ]
+
 CORE:
-  - any: ["core/**/*", "!**/*UI.scala", "!**/ui/**/*"]
-  - "common/kvstore/**/*"
-  - "common/network-common/**/*"
-  - "common/network-shuffle/**/*"
-  - "python/pyspark/*.py"
-  - "python/pyspark/tests/**/*.py"
+  - changed-files:
+    - all-globs-to-any-file: [
```

Comment:
`any` -> `all-globs-to-any-file`
The logic of `all-globs-to-any-file` has been validated in private repo using `https://github.com/panbingkun/github-action-test/pull/2` and `https://github.com/panbingkun/github-action-test/pull/3`